= ProXI: ProteomeXchange Expression Interface Specification
:sectnums:
:toc: left
:doctype: book
//only works on some backends, not HTML
:showcomments:
//use style like Section 1 when referencing within the document.
:xrefstyle: short
:figure-caption: Figure
:pdf-page-size: A4

//GitHub specific settings
ifdef::env-github[]
:tip-caption: :bulb:
:note-caption: :information_source:
:important-caption: :heavy_exclamation_mark:
:caution-caption: :fire:
:warning-caption: :warning:
endif::[]

//disable section numbering
:!sectnums:
[preface]
== Preface
_Status of This Document_

This document presents the final specification of the ProXI (The ProteomeXchange Expression Interface)
an API specification developed by members of the Human Proteome Organisation (HUPO)
Proteomics Standards Initiative (PSI) and ProteomeXchange partners. Distribution is unlimited.

_Version of This Document_

The current version of this document is: version 1.0.0-draft, Apr 2018.

The latest draft version of this document may be found at
https://github.com/bigbio/proxi-schemas.

[abstract]
[[abstract]]
== Abstract

The Human Proteome Organisation (HUPO) Proteomics Standards Initiative (PSI)
and the ProteomeXchange Consortium (ProteomeXchange) define community standards
for data representation in proteomics to facilitate data comparison, exchange
and verification.

In this context, the two organizations are working together on a shared standard
for downstream results, following mass spectrometry (MS) analysis. This document
defines a API specification to report proteomics results.

//reenable section numbering
:sectnums:
[[introduction]]
== Introduction

[[background]]
=== Background

This document addresses the systematic description of peptide/protein
identification and quantification data retrieved from mass spectrometry (MS)-based
experiments in public proteomics databases

ProXI is intended as a common interface across all major ProteomeXchange
partners to exchange peptide/protein evidences
deposited in public proteomics databases.

ProXI has been developed with a view to support the following general tasks
(more specific use cases are provided in <<use-cases-for-mztab>>):

1.  _Facilitate the sharing of final experimental results deposited in public databases.
2.  _Facilitate the interaction of public proteomics databases with other omics resources such as metabolomics and gene expression databases
3.  _Act as an output format of (web-) services_ that report MS-based results and thus can produce standardized result pages.
4.  _Be able to link to the external experimental evidence_ e.g. by referencing back to spectra files.

This document presents a specification, not a tutorial. As such, the presentation of technical details is deliberately direct.
The role of the text is to describe the model and justify design decisions made. The document does not discuss how the models should be used in practice,
consider tool support for data capture or storage, or provide comprehensive examples of the models in use. It is anticipated that tutorial material will be developed
independently of this specification.

[[document-structure]]
=== Document Structure

The remainder of this document is structured as follows. <<use-cases-for-mztab>> lists use cases ProXI
is designed to support.
<<notational-conventions>> describes the terminology used.
<<relationships-to-other-specifications>> describes how the specification presented
in <<format-specification>> relates to other specifications, both those that it extends and
those that it is intended to complement. <<resolved-design-and-scope-issues>>
discusses the reasoning behind several design decisions taken.
<<format-specification>> contains the documentation of the file. <<non-supported-use-cases>> lists use cases
that are currently not supported. Conclusions are presented in <<conclusions>>.

[[use-cases-for-mztab]]
== Use Cases for mzTab

The following cases of usage have driven the development of the ProXI data model,
and are used to define the scope of the format in version 1.0.

1. ProXI API should be simple enough to make proteomics results accessible to people outside the respective
fields. This should facilitate the sharing of data beyond the borders of the fields and make it accessible
to non-experts.

2. ProXI should contain sufficient information to provide biological insights of all findings in a proteomics
study.

3. ProXI should enable reporting at different levels of detail: ranging from a simple summary of the final
results to a detailed reporting including the experimental design.

4. It should be possible to query a ProXI implementation and retrieve which type of information is provided by the
resource and which methods are implemented. The API should be self discoverable.

5. ProXI API should be possible to retrieve information about independent experiment as long as agregation analysis of
multiple experiments.


[[notational-conventions]]
== Notational Conventions

The key words “MUST,” “MUST NOT,” “REQUIRED,” “SHALL,” “SHALL NOT,” “SHOULD,” “SHOULD NOT,” “RECOMMENDED,”
“MAY,” and “OPTIONAL” are to be interpreted as described in RFC-2119 <<bradner-1997, (Bradner 1997)>>.

The current (_v3_) ProteomeXchange definition for projects has defined the following fields:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<ProteomeXchangeDataset id="PXD008955" formatVersion="1.3.0" xsi:noNamespaceSchemaLocation="proteomeXchange-1.3.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <CvList>
        <Cv fullName="PSI-MS" uri="https://raw.githubusercontent.com/HUPO-PSI/psi-ms-CV/master/psi-ms.obo" id="MS"/>
        <Cv fullName="PRIDE" uri="https://raw.githubusercontent.com/PRIDE-Utilities/pride-ontology/master/pride_cv.obo" id="PRIDE"/>
        <Cv fullName="PSI-MOD" uri="https://raw.githubusercontent.com/MICommunity/psidev/master/psi/mod/data/PSI-MOD.obo" id="MOD"/>
        <Cv fullName="UNIMOD" uri="http://www.unimod.org/obo/unimod.obo" id="UNIMOD"/>
    </CvList>
    <DatasetSummary announceDate="2018-04-18" hostingRepository="PRIDE" title="Protein quality control by MARCH6/TRC8">
        <Description>Forward genetic screens in human cells we find that the proteasome-mediated degradation of the soluble misfolded reporter, mCherry-CL1, involves two ER-resident E3 ligases, MARCH6 and TRC8. To identify a more physiological correlate we used quantitative mass spectrometry and found that TRC8 and MARCH6 depletion altered the turnover of the tail-anchored protein Heme-Oxygenase-1 (HO-1).</Description>
        <ReviewLevel>
            <cvParam cvRef="PRIDE" accession="PRIDE:0000414" name="Peer-reviewed dataset"/>
        </ReviewLevel>
        <RepositorySupport>
            <cvParam cvRef="PRIDE" accession="PRIDE:0000417" name="Unsupported dataset by repository"/>
        </RepositorySupport>
    </DatasetSummary>
    <DatasetIdentifierList>
        <DatasetIdentifier>
            <cvParam cvRef="MS" accession="MS:1001919" name="ProteomeXchange accession number" value="PXD008955"/>
        </DatasetIdentifier>
    </DatasetIdentifierList>
    <DatasetOriginList>
        <DatasetOrigin>
            <cvParam cvRef="PRIDE" accession="PRIDE:0000402" name="Original data"/>
        </DatasetOrigin>
    </DatasetOriginList>
    <SpeciesList>
        <Species>
            <cvParam cvRef="MS" accession="MS:1001467" name="taxonomy: NCBI TaxID" value="9606"/>
        </Species>
    </SpeciesList>
    <InstrumentList>
        <Instrument id="Instrument_1">
            <cvParam cvRef="MS" accession="MS:1002416" name="Orbitrap Fusion"/>
        </Instrument>
    </InstrumentList>
    <ModificationList>
         <cvParam cvRef="MOD" accession="MOD:00397" name="iodoacetamide derivatized residue"/>
    </ModificationList>
    <ContactList>
        <Contact id="project_submitter">
            <cvParam cvRef="MS" accession="MS:1000586" name="contact name" value="James Williamson"/>
            <cvParam cvRef="MS" accession="MS:1000589" name="contact email" value="jcw76@cam.ac.uk"/>
            <cvParam cvRef="MS" accession="MS:1000590" name="contact affiliation" value="University of Cambridge"/>
            <cvParam cvRef="MS" accession="MS:1002037" name="dataset submitter"/>
        </Contact>
        <Contact id="project_lab_head">
            <cvParam cvRef="MS" accession="MS:1002332" name="lab head"/>
            <cvParam cvRef="MS" accession="MS:1000586" name="contact name" value="James Nathan"/>
            <cvParam cvRef="MS" accession="MS:1000589" name="contact email" value="jan33@cam.ac.uk"/>
            <cvParam cvRef="MS" accession="MS:1000590" name="contact affiliation" value="Cambridge Institute for Medical Research,  Department of Medicine,  University of Cambridge,  Cambridge,  CB2 0XY,  UK"/>
        </Contact>
    </ContactList>
    <PublicationList>
        <Publication id="PMID29519897">
            <cvParam cvRef="MS" accession="MS:1000879" name="PubMed identifier" value="29519897"/>
            <cvParam cvRef="PRIDE" accession="PRIDE:0000400" name="Reference" value="Stefanovic-Barrett S, Dickson AS, Burr SP, Williamson JC, Lobb IT, van den Boomen DJ, Lehner PJ, Nathan JA. MARCH6 and TRC8 facilitate the quality control of cytosolic and tail-anchored proteins. EMBO Rep. 2018"/>
        </Publication>
    </PublicationList>
    <KeywordList>
        <cvParam cvRef="MS" accession="MS:1001925" name="submitter keyword" value="Protein quality control, MARCH6, TRC8, Intramembrane proteolysis, ERAD"/>
        <cvParam cvRef="MS" accession="MS:1001926" name="curator keyword" value="Biological"/>
    </KeywordList>
    <FullDatasetLinkList>
        <FullDatasetLink>
            <cvParam cvRef="PRIDE" accession="PRIDE:0000411" name="Dataset FTP location" value="ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2018/04/PXD008955"/>
        </FullDatasetLink>
        <FullDatasetLink>
            <cvParam cvRef="MS" accession="MS:1001930" name="PRIDE project URI" value="http://www.ebi.ac.uk/pride/archive/projects/PXD008955"/>
        </FullDatasetLink>
    </FullDatasetLinkList>
    <DatasetFileList>
        <DatasetFile id="FILE_0" name="JN-SSB_Knockouts_TMT10_20pc_F1.raw">
            <cvParam cvRef="PRIDE" accession="PRIDE:0000404" name="Associated raw file URI" value="ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2018/04/PXD008955/JN-SSB_Knockouts_TMT10_20pc_F1.raw"/>
        </DatasetFile>
        <DatasetFile id="FILE_1" name="JN-SSB_Knockouts_TMT10_20pc_F10.raw">
            <cvParam cvRef="PRIDE" accession="PRIDE:0000404" name="Associated raw file URI" value="ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2018/04/PXD008955/JN-SSB_Knockouts_TMT10_20pc_F10.raw"/>
        </DatasetFile>
    </DatasetFileList>
    <RepositoryRecordList>
        <RepositoryRecord name="Protein quality control by MARCH6/TRC8" label="PRIDE project" recordID="PXD008955" repositoryID="PRIDE" uri="http://www.ebi.ac.uk/pride/archive/projects/PXD008955"/>
    </RepositoryRecordList>
</ProteomeXchangeDataset>
```

The PROXI dataset (_v1_) is defined by the following properties:

 | Field          | Description                            | Type         | Multiplicity   |
 |----------------|----------------------------------------|--------------|----------------|
 | accession      | The accession of the dataset.          | string       | 1              |
 | title          | Dataset Title                          | string       | 1              |
 | summary        | Dataset description                    | string       | 1              |
 | organisms      | List of organisms                      | OntologyTerm | 1..*           |
 | instruments    | List of instruments                    | OntologyTerm | 1..*           |
 | modifications  | List of Postranslational modifications | OntologyTerm | 1..*           |
 | contacts       | List of contacts                       | Contact      | 1..*           |

 The attributes _organisms_ , _instruments_, _modifications_ are Ontology based terms, to read more about that
 check the [_OntologyTerm_ definition](). _Contacts_ are build using the following structure:

  | Field          | Description                            | Type         | Multiplicity   |
  |----------------|----------------------------------------|--------------|----------------|
  | contactType    | Contact type (Submitter, LabHead)      | string       | 1              |
  | attributes     | List of attributes of the contact      | OntologyTerm | 1..*           |
