= ProXI: ProteomeXchange Expression Interface Specification
:sectnums:
:toc: left
:doctype: book
//only works on some backends, not HTML
:showcomments:
//use style like Section 1 when referencing within the document.
:xrefstyle: short
:figure-caption: Figure
:pdf-page-size: A4

//GitHub specific settings
ifdef::env-github[]
:tip-caption: :bulb:
:note-caption: :information_source:
:important-caption: :heavy_exclamation_mark:
:caution-caption: :fire:
:warning-caption: :warning:
endif::[]

//disable section numbering
:!sectnums:
[preface]
== Preface
_Status of This Document_

This document presents the final specification of the ProXI (The ProteomeXchange Expression Interface)
an API specification developed by members of the Human Proteome Organisation (HUPO)
Proteomics Standards Initiative (PSI) and ProteomeXchange partners. Distribution is unlimited.

_Version of This Document_

The current version of this document is: version 1.0.0-draft, Apr 2018.

The latest draft version of this document may be found at
https://github.com/bigbio/proxi-schemas.

[abstract]
[[abstract]]
== Abstract

The Human Proteome Organisation (HUPO) Proteomics Standards Initiative (PSI)
and the ProteomeXchange Consortium (ProteomeXchange) define community standards
for data representation in proteomics to facilitate data comparison, exchange
and verification.

In this context, the two organizations are working together on a shared standard
for downstream results, following mass spectrometry (MS) analysis. This document
defines a API specification to report proteomics results.

//reenable section numbering
:sectnums:
[[introduction]]
== Introduction

[[background]]
=== Background

This document addresses the systematic description of peptide/protein
identification and quantification data retrieved from mass spectrometry (MS)-based
experiments in public proteomics databases

ProXI is intended as a common interface across all major ProteomeXchange
partners to exchange peptide/protein evidences
deposited in public proteomics databases.

ProXI has been developed with a view to support the following general tasks
(more specific use cases are provided in <<use-cases-for-mztab>>):

1.  _Facilitate the sharing of final experimental results deposited in public databases.
2.  _Facilitate the interaction of public proteomics databases with other omics resources such as metabolomics and gene expression databases
3.  _Act as an output format of (web-) services_ that report MS-based results and thus can produce standardized result pages.
4.  _Be able to link to the external experimental evidence_ e.g. by referencing back to spectra files.

This document presents a specification, not a tutorial. As such, the presentation of technical details is deliberately direct.
The role of the text is to describe the model and justify design decisions made. The document does not discuss how the models should be used in practice,
consider tool support for data capture or storage, or provide comprehensive examples of the models in use. It is anticipated that tutorial material will be developed
independently of this specification.

[[document-structure]]
=== Document Structure

The remainder of this document is structured as follows. <<use-cases-for-mztab>> lists use cases ProXI
is designed to support.
<<notational-conventions>> describes the terminology used.
<<relationships-to-other-specifications>> describes how the specification presented
in <<format-specification>> relates to other specifications, both those that it extends and
those that it is intended to complement. <<resolved-design-and-scope-issues>>
discusses the reasoning behind several design decisions taken.
<<format-specification>> contains the documentation of the file. <<non-supported-use-cases>> lists use cases
that are currently not supported. Conclusions are presented in <<conclusions>>.

[[use-cases-for-mztab]]
== Use Cases for mzTab

The following cases of usage have driven the development of the ProXI data model,
and are used to define the scope of the format in version 1.0.

1. ProXI API should be simple enough to make proteomics results accessible to people outside the respective
fields. This should facilitate the sharing of data beyond the borders of the fields and make it accessible
to non-experts.

2. ProXI should contain sufficient information to provide biological insights of all findings in a proteomics
study.

3. ProXI should enable reporting at different levels of detail: ranging from a simple summary of the final
results to a detailed reporting including the experimental design.

4. It should be possible to query a ProXI implementation and retrieve which type of information is provided by the
resource and which methods are implemented. The API should be self discoverable.

5. ProXI API should be possible to retrieve information about independent experiment as long as agregation analysis of
multiple experiments.

[[api-standard-openapi]]
== API standard (OpenAPI)

The OpenAPI Specification (OAS) defines a standard, language-agnostic interface to
RESTful APIs which allows both humans and computers to discover and understand the
capabilities of the service without access to source code, documentation, or through
network traffic inspection. When properly defined, a consumer can understand and
interact with the remote service with a minimal amount of implementation logic.

The OpenAPI specification can be read here https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md

[[notational-conventions]]
== Notational Conventions

The key words “MUST,” “MUST NOT,” “REQUIRED,” “SHALL,” “SHALL NOT,” “SHOULD,” “SHOULD NOT,” “RECOMMENDED,”
“MAY,” and “OPTIONAL” are to be interpreted as described in RFC-2119 <<bradner-1997, (Bradner 1997)>>.

[[naming-notation-conventions]]
== Notation and Naming Conventions

1. A **resource** SHOULD be represented as a collection. For example, “datasets”
is a collection of datasets where we can identify a single “dataset” using the
resource URI “/datasets/{customerId}”.

2. MUST NOT use trailing forward slash (/) in URIs:

```
  http://api.example.com/device-management/managed-devices/
  http://api.example.com/device-management/managed-devices 	  // This is much better version
```

3. MUST NOT use underscores ( _ ), instead use ( - ):

It’s possible to use an underscore in place of hyphen to be used as separator – But depending on the application’s font,
it’s possible that the underscore (_) character can either get partially obscured or completely hidden in some browsers
or screens.

```
http://api.example.com/inventory-management/managed-entities/{id}/install-script-location  //More readable
http://api.example.com/inventory_management/managed_entities/{id}/install_script_location  //More error prone
```

4. SHOULD NOT Use uppercase letters in URIs:

When convenient, lowercase letters should be consistently preferred in URI paths.
**RFC 3986** defines URIs as case-sensitive except for the scheme and host components. e.g.

```
http://api.example.org/resource/1        //Better representation
http://api.example.org/Resource/1
```

5. MUST NOT use file extensions

File extensions look bad and do not add any advantage. Removing them decrease the
length of URIs as well. No reason to keep them.

```
http://api.example.com/device-management/managed-devices.xml  /*Do not use it*/
http://api.example.com/device-management/managed-devices 	/*This is correct URI*/
```

6. MUST NOT use CRUD function names in URIs

URIs MUST NOT be used to indicate that a CRUD function is performed.
URIs should be used to uniquely identify resources and not any action upon them.
HTTP request methods (headers) should be used to indicate which CRUD function is performed.

```
HTTP GET http://api.example.com/device-management/managed-devices            //Get all devices
HTTP GET http://api.example.com/device-management/managed-devices/{id}       //Get device for given Id
```

7. We RECOMMENDED to use query component to filter URI collection

Many times, you will come across requirements where you will need a collection of resources sorted,
filtered or limited based on some certain resource attribute.

For this, do not create new APIs – rather enable sorting, filtering and pagination capabilities in resource
collection API and pass the input parameters as query parameters:

```
http://api.example.com/device-management/managed-devices
http://api.example.com/device-management/managed-devices?region=USA
http://api.example.com/device-management/managed-devices?region=USA&brand=XYZ
http://api.example.com/device-management/managed-devices?region=USA&brand=XYZ&sort=installation-dateRECOMMENDED
```

[[relationship-to-other-specifications]]
== Relationship to Other Specifications

The specification described in this document has not been developed in isolation; indeed, it is designed to be complementary to, and thus used in conjunction with, several existing and emerging models. Related specifications include the following:

1.  _mzTab_ (http://www.psidev.info/mztab). mzTab files MAY be used in conjunction with peak lists, although it will be possible to use mzTab with other formats of mass spectra. This document does not assume familiarity with mzML.


[[format-specification]]
== Format specification

This section describes the structure of ProXI API implementation.

* *Field separator* +
The column delimiter is the Unicode Horizontal Tab character (Unicode codepoint 0009).
* *File encoding* +
The UTF-8 encoding of the Unicode character set is the preferred encoding for mzTab files. However, parsers should be able to recognize commonly used encodings.
* *Case sensitivity* +
All column labels and field names are case-sensitive.
* *Line prefix* +
Every line in an mzTab file MUST start with a three letter code identifying the type of line delimited by a Tab character. The three letter codes are as follows:
** `MTD` for metadata
** `SMH` for small molecule table header line (the column labels)
** `SML` for rows of the small molecule table
** `SFH` for small molecule feature header line
** `SMF` for rows of the small molecule feature table
** `SHE` for small molecule evidence header line
** `SME` for rows of the small molecule evidence table
** `COM` for comment lines

* *Header lines* +
Each table based section (protein, peptide, PSM and small molecule) MUST start with the corresponding header line. These header lines MUST only occur once in the document since each section also MUST only occur once.
* *Dates* +
Dates and times MUST be supplied in the ISO 8601 format (“YYYY-MM-DD”, “YYYY-MM-DDTHH:MMZ” respectively).
* *Decimal separator* +
In mzTab files the dot (“.”) MUST be used as decimal separator. Thousand separators MUST NOT be used in mzTab files.
* *Comment lines and empty lines* +
Comment lines can be placed anywhere in an mzTab file. These lines must start with the three-letter code COM and are ignored by most parsers. Empty lines can also occur anywhere in an mzTab file and are ignored.
* *Params* +
mzTab makes use of CV parameters. As mzTab is expected to be used in several experimental environments where parameters might not yet be available for the generated scores etc. all parameters can either report CV parameters or user parameters that only contain a name and a value. +
Parameters are always reported as `[CV label, accession, name, value]`. Any field that is not available MUST be left empty. +
 +
....
[MS, MS:1001477, SpectraST,]
[,,A user parameter, The value]
....

In case, the name of the param contains commas, quotes MUST be added to avoid problems with the parsing: `[label, accession, “first part of the param name, second part of the name”, value]`.
....
[MOD, MOD:00648, “N,O-diacetylated L-serine”,]
....

* *Sample IDs* +
To be able to supply metadata specific to each sample, ids in the format `sample[1-n]` are used. +
 +
....
MTD sample[1]-species[1] [NEWT, 9606, Homo sapiens (Human), ]
....
* *Assay IDs* +
To be able to supply metadata specific to each assay, ids in the format `assay[1-n]` are used. +
 +
....
MTD assay[1] first assay description
....
* *Study variable IDs* +
To be able to supply metadata specific to each study variable (grouping of assays), ids in the format `study_variable[1-n]` are used.
....
MTD study_variable[1]-description Group B (spike-in 0.74 fmol/uL)
....
[[sections]]
=== Sections

mzTab-M files MUST have one Metadata (MTD) section and one Small Molecule (SML) Section. In practice, we expect that most files SHOULD also include one Small Molecule Feature (SMF) section, and one Small Molecule Evidence (SME) Section. Files lacking SMF and SME sections can only present summary data about quantified molecules, without any evidence trail for how those values were derived. It will be left to reading software to determine whether additional validation will be requested such that SMF and SME tables MUST be present.

[[metadata-section]]
=== Metadata Section

The current (_v3_) ProteomeXchange definition for projects has defined the following fields:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<ProteomeXchangeDataset id="PXD008955" formatVersion="1.3.0" xsi:noNamespaceSchemaLocation="proteomeXchange-1.3.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <CvList>
        <Cv fullName="PSI-MS" uri="https://raw.githubusercontent.com/HUPO-PSI/psi-ms-CV/master/psi-ms.obo" id="MS"/>
        <Cv fullName="PRIDE" uri="https://raw.githubusercontent.com/PRIDE-Utilities/pride-ontology/master/pride_cv.obo" id="PRIDE"/>
        <Cv fullName="PSI-MOD" uri="https://raw.githubusercontent.com/MICommunity/psidev/master/psi/mod/data/PSI-MOD.obo" id="MOD"/>
        <Cv fullName="UNIMOD" uri="http://www.unimod.org/obo/unimod.obo" id="UNIMOD"/>
    </CvList>
    <DatasetSummary announceDate="2018-04-18" hostingRepository="PRIDE" title="Protein quality control by MARCH6/TRC8">
        <Description>Forward genetic screens in human cells we find that the proteasome-mediated degradation of the soluble misfolded reporter, mCherry-CL1, involves two ER-resident E3 ligases, MARCH6 and TRC8. To identify a more physiological correlate we used quantitative mass spectrometry and found that TRC8 and MARCH6 depletion altered the turnover of the tail-anchored protein Heme-Oxygenase-1 (HO-1).</Description>
        <ReviewLevel>
            <cvParam cvRef="PRIDE" accession="PRIDE:0000414" name="Peer-reviewed dataset"/>
        </ReviewLevel>
        <RepositorySupport>
            <cvParam cvRef="PRIDE" accession="PRIDE:0000417" name="Unsupported dataset by repository"/>
        </RepositorySupport>
    </DatasetSummary>
    <DatasetIdentifierList>
        <DatasetIdentifier>
            <cvParam cvRef="MS" accession="MS:1001919" name="ProteomeXchange accession number" value="PXD008955"/>
        </DatasetIdentifier>
    </DatasetIdentifierList>
    <DatasetOriginList>
        <DatasetOrigin>
            <cvParam cvRef="PRIDE" accession="PRIDE:0000402" name="Original data"/>
        </DatasetOrigin>
    </DatasetOriginList>
    <SpeciesList>
        <Species>
            <cvParam cvRef="MS" accession="MS:1001467" name="taxonomy: NCBI TaxID" value="9606"/>
        </Species>
    </SpeciesList>
    <InstrumentList>
        <Instrument id="Instrument_1">
            <cvParam cvRef="MS" accession="MS:1002416" name="Orbitrap Fusion"/>
        </Instrument>
    </InstrumentList>
    <ModificationList>
         <cvParam cvRef="MOD" accession="MOD:00397" name="iodoacetamide derivatized residue"/>
    </ModificationList>
    <ContactList>
        <Contact id="project_submitter">
            <cvParam cvRef="MS" accession="MS:1000586" name="contact name" value="James Williamson"/>
            <cvParam cvRef="MS" accession="MS:1000589" name="contact email" value="jcw76@cam.ac.uk"/>
            <cvParam cvRef="MS" accession="MS:1000590" name="contact affiliation" value="University of Cambridge"/>
            <cvParam cvRef="MS" accession="MS:1002037" name="dataset submitter"/>
        </Contact>
        <Contact id="project_lab_head">
            <cvParam cvRef="MS" accession="MS:1002332" name="lab head"/>
            <cvParam cvRef="MS" accession="MS:1000586" name="contact name" value="James Nathan"/>
            <cvParam cvRef="MS" accession="MS:1000589" name="contact email" value="jan33@cam.ac.uk"/>
            <cvParam cvRef="MS" accession="MS:1000590" name="contact affiliation" value="Cambridge Institute for Medical Research,  Department of Medicine,  University of Cambridge,  Cambridge,  CB2 0XY,  UK"/>
        </Contact>
    </ContactList>
    <PublicationList>
        <Publication id="PMID29519897">
            <cvParam cvRef="MS" accession="MS:1000879" name="PubMed identifier" value="29519897"/>
            <cvParam cvRef="PRIDE" accession="PRIDE:0000400" name="Reference" value="Stefanovic-Barrett S, Dickson AS, Burr SP, Williamson JC, Lobb IT, van den Boomen DJ, Lehner PJ, Nathan JA. MARCH6 and TRC8 facilitate the quality control of cytosolic and tail-anchored proteins. EMBO Rep. 2018"/>
        </Publication>
    </PublicationList>
    <KeywordList>
        <cvParam cvRef="MS" accession="MS:1001925" name="submitter keyword" value="Protein quality control, MARCH6, TRC8, Intramembrane proteolysis, ERAD"/>
        <cvParam cvRef="MS" accession="MS:1001926" name="curator keyword" value="Biological"/>
    </KeywordList>
    <FullDatasetLinkList>
        <FullDatasetLink>
            <cvParam cvRef="PRIDE" accession="PRIDE:0000411" name="Dataset FTP location" value="ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2018/04/PXD008955"/>
        </FullDatasetLink>
        <FullDatasetLink>
            <cvParam cvRef="MS" accession="MS:1001930" name="PRIDE project URI" value="http://www.ebi.ac.uk/pride/archive/projects/PXD008955"/>
        </FullDatasetLink>
    </FullDatasetLinkList>
    <DatasetFileList>
        <DatasetFile id="FILE_0" name="JN-SSB_Knockouts_TMT10_20pc_F1.raw">
            <cvParam cvRef="PRIDE" accession="PRIDE:0000404" name="Associated raw file URI" value="ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2018/04/PXD008955/JN-SSB_Knockouts_TMT10_20pc_F1.raw"/>
        </DatasetFile>
        <DatasetFile id="FILE_1" name="JN-SSB_Knockouts_TMT10_20pc_F10.raw">
            <cvParam cvRef="PRIDE" accession="PRIDE:0000404" name="Associated raw file URI" value="ftp://ftp.pride.ebi.ac.uk/pride/data/archive/2018/04/PXD008955/JN-SSB_Knockouts_TMT10_20pc_F10.raw"/>
        </DatasetFile>
    </DatasetFileList>
    <RepositoryRecordList>
        <RepositoryRecord name="Protein quality control by MARCH6/TRC8" label="PRIDE project" recordID="PXD008955" repositoryID="PRIDE" uri="http://www.ebi.ac.uk/pride/archive/projects/PXD008955"/>
    </RepositoryRecordList>
</ProteomeXchangeDataset>
```

The PROXI dataset (_v1_) is defined by the following properties:

 | Field          | Description                            | Type         | Multiplicity   |
 |----------------|----------------------------------------|--------------|----------------|
 | accession      | The accession of the dataset.          | string       | 1              |
 | title          | Dataset Title                          | string       | 1              |
 | summary        | Dataset description                    | string       | 1              |
 | organisms      | List of organisms                      | OntologyTerm | 1..*           |
 | instruments    | List of instruments                    | OntologyTerm | 1..*           |
 | modifications  | List of Postranslational modifications | OntologyTerm | 1..*           |
 | contacts       | List of contacts                       | Contact      | 1..*           |

 The attributes _organisms_ , _instruments_, _modifications_ are Ontology based terms, to read more about that
 check the [_OntologyTerm_ definition](). _Contacts_ are build using the following structure:

  | Field          | Description                            | Type         | Multiplicity   |
  |----------------|----------------------------------------|--------------|----------------|
  | contactType    | Contact type (Submitter, LabHead)      | string       | 1              |
  | attributes     | List of attributes of the contact      | OntologyTerm | 1..*           |
